### DISCUSSION TITLE
USB Device Composite Class apps are now available!

### DISCUSSION THREAD

**Discussion by ALABSTM** [STMicroelectronics/STM32Cube_MCU_Overall_Offer#18] [Announcements]
## USB Device Composite Class apps are now available!

STM32 Classic Core USB Device Composite Class apps are now available on a [dedicated branch](https://github.com/STMicroelectronics/STM32CubeH7/tree/dev/usb/composite/Projects/STM32H743I-EVAL/Applications/USB_Device) `dev/usb/composite` in the `STM32CubeH7` repo.

**Four** new applications are proposed:
1. Composite **Audio and HID** classes
2. Composite **CDC and HID** classes
3. Composite **CDC and Custom HID** classes
4. Composite **CDC and Printer** classes

Provided for the `STM32H743I-EVAL` board, these applications could be **easily tailored** for other boards.
Compatible with versions **2.10.0** and **2.11.0** of ST's USB Device library, they are also **backward compatible** with older versions of the library.

Enjoy our **STM32Cube** offer and stay tuned!
##
**Known limitations:** App 'Composite_Audio_HID' would fail at link-time since files under `./Middlewares/ST/STM32_Audio` have been removed as subject to "click-thru" license.

**Comment by SloBloLabs** [comment]
Hi @ALABSTM, I am trying to implement a composite device combining DFU, CDC and AUDIO (to become Midi later) using USB Device Library 2.11.1, but no success so far. Is there any documentation around how to use the library as intended? Find a full report of my status of attempt here: https://community.st.com/s/feed/0D53W00002G7LqjSAF. I would be glad to get in touch and learn something.

**Reply by ALABSTM** [reply]
Hi @SloBloLabs,

From what I read of your post on the ST Community, I noticed you could overcome the issue you were facing. Congratulations.

Regarding this channel, it is rather dedicated to report bugs or enhancement proposals to the software published within the different repositories. A support request such as yours would have been kindly redirected to the ST Community.

Thank you for your comprehension and, once again, glad you could successfully implement your composite device-based application.

With regards,

**Comment by SloBloLabs** [comment]
Hi Ali,

 

nice to hear from and glad to see my contribution to the ST forum found your attention.

 

I am a little unsure how to process your input, though:

 

Shell I post such requests as mine to different location and if yes, where to?

Or would you like to incorporate my solution to some repo that would be available for the community for reuse?


Kind regards,

Oliver Walczak

 
 

Gesendet: Montag, 12. Juni 2023 um 12:24 Uhr
Von: "Ali LABBENE" ***@***.***>
An: "STMicroelectronics/STM32Cube_MCU_Overall_Offer" ***@***.***>
Cc: "SloBloLabs" ***@***.***>, "Mention" ***@***.***>
Betreff: Re: [STMicroelectronics/STM32Cube_MCU_Overall_Offer] USB Device Composite Class apps are now available! (Discussion #18)


 

Hi @SloBloLabs,

From what I read of your post on the ST Community, I noticed you could overcome the issue you were facing. Congratulations.

Regarding this channel, it is rather dedicated to report bugs or enhancement proposals to the software published within the different repositories. A support request such as yours would have been kindly redirected to the ST Community.

Thank you for your comprehension and, once again, glad you could successfully implement your composite device-based application.

With regards,

**Comment by paloky** [comment]
Hi @ALABSTM .

I don't see how use the composite in CubeIDE Configuration device.
In another post, you say it's is included since version 1.27.0 of the STM32CubeF4 firmware.
I'm currently using version 1.27.1 of F4 and I do not see the Composite option.

It's possible to Add and example of composite CDC+MSC.
Thank's a lot.
